crates/cfg/Cargo.toml

```diff
@@ -17,3 +17,5 @@ tt = { path = "../tt", version = "0.0.0" }
 mbe = { path = "../mbe" }
 syntax = { path = "../syntax" }
 expect-test = "1.1"
+arbitrary = { version = "1", features = ["derive"] }
```

Comment:
Is anything in the crate graph using arbitrary already? If not, I suggest depending on derive_arbitrary directly: that way, I can add arbitrary to small_str without putting it after syn in compilation graph.

Ideally, we’d do that for serde as well, but asking everyone to depend on serde_derive directly feels like a too big of an ask.

Reply:
Hmm, but the generated code won't compile unless `arbitrary` is also a dependency. Maybe I misunderstand what you're suggesting though?

Reply:
Can it be a dev-dependency and the derive behind `cfg(test)`?

Reply:
Yeah, I had thought that I was being confusing. 

Basically, we can do either

```toml
arbitrary = { version = "1", features = ["derive"] }
```

or

```toml
arbitrary = { version = "1" }
derive_arbitrary = { version = "1" }
```

I suggest going with the second approach. The reason for that is that in SmolStr I want to do 

```toml
arbitrary = { version = "1" }
```

as I don't need derives.


Now, if we flip the `derive` feature for *this* crate, that would mean that `smol_str` needs to be compiled after `syn`, although it itself doesn't use derive feature. And that would mean that we put everything that depends on `smol_str` after `syn`.

That is, the core issue is that enabling `derive` in one crate might make unrelated crates slower to compile. 

In other words, if you obsess with compile times like I do, then re-exporting proc macros, like serde and arbitrary do, is an anti-pattern. Luckely, we can opt-out of that, by using derive crates manually, and I'd love to try that at least for arbitrary.

Reply:
Ah, now that makes sense!